src/test/pages/MembersPage.page.ts

```diff
@@ -5,70 +5,185 @@ import generateRandomName from "../utils/generateRandomName";
+    private optionsFeature: Locator = this.page.locator(".dropdown-menu.left-0")
+    private edtiMembershipButton: Locator = this.page.locator(".dropdown-menu .dropdown-menu-item.undefined:first-child")
+    private membershipInput: Locator = this.page.locator(".form-container > div:nth-child(2) input ")
+    private membershipContainer: Locator = this.page.locator(".current-membership-name")
+    private calendarButton: Locator = this.page.locator(".fa-calendar-day")
+    private calendarArrow: Locator = this.page.locator("button .fa-arrow-right")
+    private calendarDay: Locator = this.page.locator(".react-datepicker__day.react-datepicker__day--018")
+    private membershipHistoryRow : Locator = this.page.locator(".card table .table-row.undefined:first-child")
+
+    constructor(page: Page) {
         super(page);
-        
     }
-    name : string = generateRandomName();
-    public async clickPlusSign () {
-     await this.plus.click();
-   }
 
-    public async addFirstName ()
-    {    
-        await this.firstNameInput.fill(this.name);
+    name = faker.person.firstName()
+    memberName = faker.person.firstName()
+    membershipPlan = "abcas"
+
+    public async clickPlusSign() {
+        await this.plus.click();
     }
 
-    public async addLastName(){
-       await this.lastNameInput.fill(generateRandomName())
+    public async addFirstName() {
+        await this.firstNameInput.fill(this.name);
     }
 
-    public async addPhoneNumber(){
-         this.phoneNumberInput.click();
-         this.phoneNumberInput.type(generateRandomPhoneNumber())
+    public async addLastName() {
+        await this.lastNameInput.fill(faker.person.firstName())
     }
 
-    public async addEmail(){
-          this.emailInput.fill(generateRandomEmail());
-          await new Promise(resolve => setTimeout(resolve, 4000));
+    public async addPhoneNumber() {
+        this.phoneNumberInput.click();
+        this.phoneNumberInput.type(faker.phone.number('004######'))
     }
-    
-   public async clickSaveButton(){
-      await  this.saveButton.click();
-    
-      await new Promise(resolve => setTimeout(resolve, 4000));
 
-   }
-    public async addDate(){
-       
+    public async addEmail() {
+        this.emailInput.fill(generateRandomEmail());
+        await new Promise(resolve => setTimeout(resolve, 4000));
     }
 
-    public async selectMembership(){
+    public async clickSaveButton() {
+        await this.saveButton.click();
+        await new Promise(resolve => setTimeout(resolve, 4000));
+    }
+    public async addDate() {
+    }
+    public async selectMembership() {
         this.membershipSelector.click()
     }
 
-   public async checkAddedMember() { 
-      
-       const element = await this.page.waitForSelector(`text=${this.name}`);
-       expect(element).not.toBeNull(); 
+    public async checkAddedMember() {
+        const element = await this.page.waitForSelector(`text=${this.name}`);
+        expect(element).not.toBeNull();
```

Comment:
It's a good practice to add detailed messages in case if expect condition fails. It will be easier to read stack trace.
 For example, 
" expect(this.city, 'Validating city').to.eql(dynamoPayload.city)"